Sources/KlaviyoLocation/KlaviyoLocationManager.swift

```diff
@@ -69,23 +69,36 @@ class KlaviyoLocationManager: NSObject {
             }
             return
         }
-        let remoteGeofences = await geofenceService.fetchGeofences(apiKey: apiKey)
+
+        // Calculate available spots for geofences based on iOS limit
         let activeGeofences = await getActiveGeofences()
+        let availableSpots = 20 - locationManager.monitoredRegions.count + activeGeofences.count
+
+        if #available(iOS 14.0, *) {
+            Logger.geoservices.info("📊 Available spots for geofences: \(availableSpots)")
+        }
+
+        let (latitude, longitude) = transformCoordinates(locationManager.location?.coordinate)
+        var remoteGeofences = await geofenceService.fetchGeofences(apiKey: apiKey, latitude: latitude, longitude: longitude)
+
+        // Filter to nearest geofences if location is available
+        if let latitude, let longitude {
+            let nearestGeofences = GeofenceDistanceCalculator.filterToNearest(
+                geofences: remoteGeofences,
+                userLatitude: locationManager.location?.coordinate.latitude ?? latitude,
+                userLongitude: locationManager.location?.coordinate.longitude ?? longitude,
+                limit: availableSpots
+            )
+            remoteGeofences = Set(nearestGeofences)
+        }
 
         let geofencesToRemove = activeGeofences.subtracting(remoteGeofences)
         let geofencesToAdd = remoteGeofences.subtracting(activeGeofences)
+        if #available(iOS 14.0, *) {
+            Logger.geoservices.warning("⚠️ Adding \(geofencesToAdd.count) and removing \(geofencesToRemove.count) geofences")
```

Comment:
```suggestion
            Logger.geoservices.log("Adding \(geofencesToAdd.count) and removing \(geofencesToRemove.count) geofences")
```
nit: this doesn't seem to be a warning-level event to me. It's just saying "hey, we're gonna do this totally normal thing" which to me is a `log`-level event